File: medium/array/Maximum Gap.cpp
class Solution {
public:
    int maximumGap(vector<int>& nums) {
        if(nums.size() < 2) return 0; int m=0,k=0;
        sort(nums.begin(),nums.end());
        
        for(int i=1;i<nums.size();i++){
            if(nums[i-1]<nums[i]) k = nums[i] - nums[i-1];
            else if(nums[i-1]>nums[i]) k = nums[i-1] - nums[i];
            if(k>m) m = k;
        }return m;
    }
};

File: medium/array/Maximum Gap.py
class Solution(object):
    def maximumGap(self, nums):
        nums.sort()
        if len(nums) < 2:
            return 0
        m = 0
        k=0
        for i in range(1,len(nums)):
            if nums[i]>nums[i-1]:
                k = nums[i] - nums[i-1]
            else: 
                k = nums[i-1] - nums[i]
            if k>m:
                m=k
        return m